### PR TITLE
Fix empty string handling in push notifications

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/11460

Empty strings are falsey, and the state key match for a tombstone event is an empty string. Ergo, nothing happens because all the conditions fail. 

This PR also does a small amount of drive-by cleanup.

Evidence of this working:
image

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -714,3 +714,7 @@ module.exports.ensureNoTrailingSlash = function(url) {
 module.exports.sleep = (ms, value) => new Promise((resolve => {
     setTimeout(resolve, ms, value);
 }));
+
+module.exports.isNullOrUndefined = function(val) {
```
Comment (turt2live) [inline src/utils.js]: rationale: I'm super lazy and don't feel like typing this OR condition all over the place.

File: src/pushprocessor.js [hunk 2]
```diff
@@ -268,7 +268,7 @@ function PushProcessor(client) {
         }
 
         const val = valueForDottedKey(cond.key, ev);
-        if (!val || typeof val != 'string') {
+        if (isNullOrUndefined(val) || typeof val != 'string') {
```
Comment (jryans) [inline src/pushprocessor.js]: Is the left side needed here? `typeof null === "object"` and `typeof undefined === "undefined"`...

Fine to keep if it's clearer as-is though. I guess a comment on what cases we mean to include vs. exclude would help.

Comment (turt2live) [reply src/pushprocessor.js]: Probably not, but I'm terrified of changes in this area of code (given age).

**Comment by turt2live** [comment]
I'm merging regardless of tests because the tests are flakey :|

**Pull request merged**